numpy/core/function_base.py

```diff
@@ -123,21 +123,43 @@ def linspace(start, stop, num=50, endpoint=True, retstep=False, dtype=None,
     if num < 0:
         raise ValueError("Number of samples, %s, must be non-negative." % num)
     div = (num - 1) if endpoint else num
-
-    # Convert float/complex array scalars to float, gh-3504
-    # and make sure one can use variables that have an __array_interface__, gh-6634
-    start = asanyarray(start) * 1.0
-    stop  = asanyarray(stop)  * 1.0
+    # Make sure one can use variables that have an __array_interface__, gh-6634
+    start = asanyarray(start)
+    stop  = asanyarray(stop)
+
+    # check if dealing with datetime64 or timedelta64
+    is_datelike = start.dtype.kind in 'Mm'
+    if is_datelike:
+        # For datetime/timedelta, dtype must be specified:
+        # start and stop will first be cast to this dtype.
+        # Warning: dtype must have high enough precision so
+        # that it is at all possible to draw num equispaced
+        # samples.
+        if dtype is None:
+            raise ValueError("linspace with {} objects cannot be called with"
+                    "dtype=None".format(start.dtype))
+        if np.isnat(start):
+            raise ValueError("start is NaT, linspace won't work.")
+        if np.isnat(stop):
```

Comment:
Unfortunately the spelling is `_nx` in this file. Will review more later, but this causes tests to fail.

Reply:
Thanks for the comment. I have fixed it and the testsuite now passes.